<?xml version='1.0' ?>                         

<launch>                                       

  <arg name="subns" default="yin" description="Names the composition of nodes"/>
  <arg name="buddy_subns" default="yang" descrption="Names the buddy composition of nodes" />
  <arg name="active_node" default="true" description="Indicates if this is the active node"/>
  <arg name="namespace" default="talker" description="Name of the namespace that contains all the nodes "/>
  <arg name="verbose" default="true" description="Extra verbosity" />

  <node pkg="stubborn_buddies"
        exec="linktime_composition"
        output="both"
        namespace="/$(var namespace)/$(var subns)">

    <param name="active_node" value="$(var active_node)" />
    <param name="subns" value="$(var subns)" />
    <param name="buddy_subns" value="$(var buddy_subns)" />
    <param name="namespace" value="$(var namespace)" />
    <param name="verbose" value="$(var verbose)" />
    <param name="run_composition_command" value="ros2 launch stubborn_buddies composition.launch.xml active_node:=false namespace:=$(var namespace) verbose:=$(var verbose) buddy_subns:=$(var subns) subns:=$(var buddy_subns) &amp;" />
           
  </node>

</launch>
